doc/api/fs.md

```diff
@@ -1101,22 +1101,171 @@ changes:
 * Returns: {AsyncIterator} An AsyncIterator that yields the paths of files
   that match the pattern.
 
+Retrieves the file paths matching the specified pattern(s).
+
+#### Pattern Syntax
+
+The glob implementation supports the following pattern syntax:
+
+**Basic Wildcards:**
+
+| Pattern | Description |
+| ------- | ----------- |
+| `*` | Matches any number of characters within a single path segment. |
+| `?` | Matches exactly one character. |
+| `[abc]` | Matches any character in the brackets. |
+| `[a-z]` | Matches any character in the range. |
+| `[!abc]` or `[^abc]` | Matches any character not in the brackets. |
+| `[!a-z]` or `[^a-z]` | Matches any character not in the range. |
+
+**Globstar:**
+
+| Pattern | Description |
+| ------- | ----------- |
+| `**` | Matches zero or more directories recursively. |
+| `**/*.js` | Matches all `.js` files in any subdirectory. |
+
+**Brace Expansion:**
+
+| Pattern | Description |
+| ------- | ----------- |
+| `{a,b,c}` | Matches any of the comma-separated patterns. |
+| `{1..5}` | Matches any number in the range (1, 2, 3, 4, 5). |
+| `*.{js,ts}` | Matches files with `.js` or `.ts` extensions. |
+
+**Extended Glob Patterns:**
+
+| Pattern | Description |
+| ------- | ----------- |
+| `+(pattern)` | Matches one or more occurrences of the pattern. |
+| `*(pattern)` | Matches zero or more occurrences of the pattern. |
+| `?(pattern)` | Matches zero or one occurrence of the pattern. |
+| `!(pattern)` | Matches anything except the pattern. |
+
+#### Examples
+
+The following examples demonstrate common glob usage patterns.
+
+**Basic pattern matching:**
+
+Match all `.js` files in the current directory.
+
 ```mjs
 import { glob } from 'node:fs/promises';
 
-for await (const entry of glob('**/*.js'))
+for await (const entry of glob('*.js'))
   console.log(entry);
 ```
 
 ```cjs
 const { glob } = require('node:fs/promises');
 
 (async () => {
-  for await (const entry of glob('**/*.js'))
+  for await (const entry of glob('*.js'))
     console.log(entry);
 })();
 ```
 
+**Recursive search with brace expansion:**
+
+Match all `.js` and `.ts` files in any subdirectory.
+
+```mjs
+import { glob } from 'node:fs/promises';
+
+for await (const entry of glob('**/*.{js,ts}'))
+  console.log(entry);
+```
+
+**Using `Array.fromAsync` to collect results:**
+
+Convert the async iterator to an array for easier manipulation.
```

Comment:
Thanks for addressing my comments!

I think the example usages can be condensed quite a bit. For example, only demonstrate how to import it once. Then you also don't have to deal with the CJS/ESM split all over the place.

What does 'for easier manipulation' mean here? Seems a bit fluff.

Maybe even just put everything in a table with only the glob pattern (inside `glob(...)` to make it more clear) and with a short description next to it.